ci/check_style.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-logger "Create checks conda environment"
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key checks \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
+
+rapids-mamba-retry env create --force -f env.yaml -n checks
+conda activate checks
+
+set +e
+
+FORMAT_FILE_URL=https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-23.02/cmake-format-rapids-cmake.json
+export RAPIDS_CMAKE_FORMAT_FILE=/tmp/rapids_cmake_ci/cmake-formats-rapids-cmake.json
+mkdir -p $(dirname ${RAPIDS_CMAKE_FORMAT_FILE})
+wget -O ${RAPIDS_CMAKE_FORMAT_FILE} ${FORMAT_FILE_URL}
+
+# TODO: enable once pre-commit checks are configured.
+# # Run pre-commit checks
+# pre-commit run --hook-stage manual --all-files --show-diff-on-failure
+
+# Run flake8 and get results/return code
+FLAKE=$(flake8 --config=python/setup.cfg)
```

Comment:
Something seems not quite right here. This config specifies `.pyx` and `.pxd` files but flake8 dropped support for non-`.py` files in version 4. That necessitated the creation of [flake8-force](https://github.com/kmaehashi/flake8-force). The style check seems to pass with `flake8 6.0.0` (because `flake8` is unpinned in `dependencies.yaml`). I see the same results locally, but I don't understand why this is "working" (or passing/skipping silently) without `flake8-force` or pinning to `flake8<4`. This isn't a blocker for this PR but I think it might be worth investigating in the pre-commit migration. cc: @csadorf